### ISSUE TITLE
Add Enum type

### ISSUE THREAD

**Issue by roblabla** [reswitched/SwIPC#24]
I added structs in #14 . Now I should add enums. Those should follow C enums: they'd probably look like

```
type Partition = enum<u32> {
    NAME1,
    NAME2,
    NAME3 = 5
};
```

Those could then be codegen's to allow prettier interfaces in each language.

**Comment by roblabla** [comment]
https://github.com/roblabla/SwIPC/commit/1a3e14c73b101261da6603016486463eea589557

**Comment by misson20000** [comment]
Closed in 9a1506a45b01de95f0f40b731ce758c1314589f5
